### DISCUSSION TITLE
微调GLM的时候为啥只用给input_ids和labels？？？

### DISCUSSION THREAD

**Discussion by ciaoyizhen** [zai-org/ChatGLM3#995] [Q&A]
ChatGLM不是使用的GLM模型吗  GLM模型的forward中不是要求给出position和attentionmask吗   但是我看了微调的代码 却只给了input_ids 这个是为啥？   是哪里实现了还是我看漏了？？   求助！！！

**Comment by ciaoyizhen** [comment]
顺便再问一下 我看了之前的ptuing的微调  好像没有给eos，但是这个版本给了 这是为啥呢 原来的有问题吗？

**Comment by zRzRzRzRzRzRzR** [comment]
直接使用了peft 调了，全部用继承的方式

**Reply by ciaoyizhen** [reply]
谢谢明白了！！！
